core/src/main/java/cucumber/api/CucumberConfiguration.java

```diff
@@ -0,0 +1,18 @@
+package cucumber.api;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
+public @interface CucumberConfiguration {
+  Type type() default Type.YAML;
+
+  String path();
+
+  public enum Type {
+    YAML, JSON, PROPERTIES, ANNOTATION;
```

Comment:
As @CucumberConfiguration` and `@CucumberOptions` are different sources of configuration and  as yaml, json and properties can be derived from the file name I don't think this type is needed.

Reply:
Makes sense. I will change it accordingly.